lightning/src/chain/keysinterface.rs

```diff
@@ -1101,11 +1107,11 @@ impl KeysInterface for KeysManager {
 	}
 
 	fn get_destination_script(&self) -> Script {
-		self.destination_script.clone()
+		bitcoin::Address::p2wpkh(&::bitcoin::PublicKey{compressed: true, key: self.shutdown_pubkey.public_key.key}, Network::Testnet).unwrap().script_pubkey()
```

Comment:
This needs to get the `channel_keys_id`-modified pubkey, not the HD root. I think we just want to pass the `channel_keys_id` parameter here as a parameter.

Reply:
So I tried changing the function signature to include the channel_keys_id but in certain places, I'm not sure how to pass that value since it does not exist there. For example in `channel.rs`  -> `new_outbound`, the value does not exist in that scope. I see that there is a `get_secure_random_bytes` in `keysinterface`, would that work instead?

Reply:
Hmm, its available there specifically as `holder_signer.channel_keys_id()`. No, we cannot just pick a random value and use that instead.